### PR TITLE
Feature/rating type edition page

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2929957&fileview=grid
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2929959&fileview=grid
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2929960&fileview=grid
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2929961&fileview=grid

### CODE REVIEW

File: src/actions/ranking-actions.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+export const RATING_TYPE_SCORE_TYPE_REMOVED = 'RATING_TYPE_SCORE_TYPE_REMOVED';
+export const RATING_TYPE_SCORE_TYPE_ADDED = 'RATING_TYPE_SCORE_TYPE_ADDED';
+export const RATING_TYPE_SCORE_TYPE_ORDER_UPDATED = 'RATING_TYPE_SCORE_TYPE_ORDER_UPDATED';
+
+export const getRatingType = (ratingTypeId) => (dispatch, getState) => {
+
+    const {loggedUserState, currentSummitState, currentSelectionPlanState} = getState();
+    const {accessToken} = loggedUserState;
+    const {currentSummit} = currentSummitState;
+    const currentSelectionPlan = currentSelectionPlanState.entity;
```
Comment (smarcet) [inline src/actions/ranking-actions.js:46]: do 
const {entity: currentSelectionPlan} = currentSelectionPlanState;

File: src/actions/ranking-actions.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+export const resetRatingTypeForm = () => (dispatch, getState) => {
+    dispatch(createAction(RESET_RATING_TYPE_FORM)({}));
+};
+
+export const saveRatingType = (entity) => (dispatch, getState) => {
+
+    const {loggedUserState, currentSummitState, currentSelectionPlanState} = getState();
+    const {accessToken} = loggedUserState;
+    const {currentSummit} = currentSummitState;
+    const currentSelectionPlan = currentSelectionPlanState.entity;
```
Comment (smarcet) [inline src/actions/ranking-actions.js:76]: same as above

File: src/actions/ranking-actions.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+    dispatch(startLoading());
+
+    const normalizedEntity = normalizeEntity(entity);
+
+    if (entity.id) {
+        putRequest(
```
Comment (smarcet) [inline src/actions/ranking-actions.js:83]: do return and avoid else

File: src/actions/ranking-actions.js [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+            });
+
+    } else {
+        const success_message = {
+            title: T.translate("general.done"),
+            html: T.translate("edit_rating_type.rating_type_created"),
+            type: 'success'
+        };
+
+        postRequest(
```
Comment (smarcet) [inline src/actions/ranking-actions.js:102]: do return here

File: src/components/forms/rating-type-form.js [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+                                 <input type="button" onClick={this.props.onAddScoreType}
+                                     className="btn btn-primary pull-right" value={T.translate("edit_rating_type.add_score_type")} />
+                            </div>
+                        </div>
+                        <SortableTable
+                            options={scoreTypesOptions}
+                            data={entity.score_types}
+                            columns={scoreTypesColumns}
+                            dropCallback={this.props.onUpdateScoreTypeOrder}
+                            orderField="order"
```
Comment (smarcet) [inline src/components/forms/rating-type-form.js:134]: order field is score

File: src/reducers/ranking/rating-type-reducer.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+    RATING_TYPE_SCORE_TYPE_ORDER_UPDATED
+} from '../../actions/ranking-actions';
+
+import { LOGOUT_USER, VALIDATE } from 'openstack-uicore-foundation/lib/actions';
+import { SET_CURRENT_SUMMIT } from '../../actions/summit-actions';
+
+export const DEFAULT_ENTITY = {
+    id          : 0,
+    name        : '',
+    weight      : 0,
```
Comment (smarcet) [inline src/reducers/ranking/rating-type-reducer.js:31]: 0.0

File: src/actions/ranking-actions.js [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+    const {accessToken} = loggedUserState;
+    const {currentSummit} = currentSummitState;
+    const currentSelectionPlan = currentSelectionPlanState.entity;
+
+    dispatch(startLoading());
+
+    const normalizedEntity = normalizeEntity(entity);
+
+    if (entity.id) {
+        putRequest(
```
Comment (smarcet) [inline src/actions/ranking-actions.js:194]: please add return

File: src/actions/ranking-actions.js [hunk 8]
```diff
@@ -0,0 +1,290 @@
...
+            });
+
+    } else {
+        const success_message = {
+            title: T.translate("general.done"),
+            html: T.translate("edit_score_type.score_type_created"),
+            type: 'success'
+        };
+
+        postRequest(
```
Comment (smarcet) [inline src/actions/ranking-actions.js:213]: please add return

File: src/actions/ranking-actions.js [hunk 9]
```diff
@@ -0,0 +1,290 @@
...
+            `${window.API_BASE_URL}/api/v1/summits/${currentSummit.id}/selection-plans/${currentSelectionPlan.id}/track-chair-rating-types/${ratingTypeId}/score-types/${entity.id}?access_token=${accessToken}`,
+            normalizedEntity,
+            authErrorHandler,
+            entity
+        )({})(dispatch)
+            .then(() => {
+                dispatch(showSuccessMessage(T.translate("edit_score_type.score_type_saved")));
+            });
+
+    } else {
```
Comment (smarcet) [inline src/actions/ranking-actions.js:206]: remove this else

File: src/actions/ranking-actions.js [hunk 10]
```diff
@@ -0,0 +1,290 @@
...
+        const success_message = {
+            title: T.translate("general.done"),
+            html: T.translate("edit_rating_type.rating_type_created"),
+            type: 'success'
+        };
+
+        postRequest(
+            createAction(UPDATE_RATING_TYPE),
```
Comment (smarcet) [inline src/actions/ranking-actions.js:103]: why u trigger this action, just set null

File: src/actions/ranking-actions.js [hunk 11]
```diff
@@ -0,0 +1,290 @@
...
+            createAction(RATING_TYPE_ADDED),
```
Comment (smarcet) [inline src/actions/ranking-actions.js:104]: this is not attended on the reducer , ( another action is attended ) causing not refreshing the selection plan list

File: src/actions/ranking-actions.js [hunk 12]
```diff
@@ -0,0 +1,290 @@
...
+    const {currentSummit} = currentSummitState;
+    const currentSelectionPlan = currentSelectionPlanState.entity;
+
+    dispatch(startLoading());
+
+    const normalizedEntity = normalizeEntity(entity);
+
+    if (entity.id) {
+        putRequest(
+            createAction(UPDATE_SCORE_TYPE),
```
Comment (smarcet) [inline src/actions/ranking-actions.js:195]: what is the purpose of this action ?

File: src/reducers/ranking/score-type-reducer.js [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+                return state;
+            } else {
+                return {...state,  entity: {...DEFAULT_ENTITY}, errors: {} };
+            }
+        }
+        case SET_CURRENT_SUMMIT:
+        case RESET_SCORE_TYPE_FORM: {
+            return {...state,  entity: {...DEFAULT_ENTITY}, errors: {} };
+        }
+        case UPDATE_SCORE_TYPE: {
```
Comment (smarcet) [inline src/reducers/ranking/score-type-reducer.js:52]: this should be done on SCORE_TYPE_UPDATED not on the initial action

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review the comments
regards

File: src/actions/ranking-actions.js (hunk 1 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:46]: done

File: src/actions/ranking-actions.js (hunk 4 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:102]: done

File: src/components/forms/rating-type-form.js (hunk 5 above)
Comment (romanetar) [reply src/components/forms/rating-type-form.js:134]: done

File: src/reducers/ranking/rating-type-reducer.js (hunk 6 above)
Comment (romanetar) [reply src/reducers/ranking/rating-type-reducer.js:31]: done

File: src/actions/ranking-actions.js (hunk 7 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:194]: done

File: src/actions/ranking-actions.js (hunk 8 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:213]: done

File: src/actions/ranking-actions.js (hunk 9 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:206]: done

File: src/actions/ranking-actions.js (hunk 10 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:103]: done

File: src/reducers/ranking/score-type-reducer.js (hunk 13 above)
Comment (romanetar) [reply src/reducers/ranking/score-type-reducer.js:52]: done

File: src/actions/ranking-actions.js (hunk 12 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:195]: removed

File: src/actions/ranking-actions.js (hunk 11 above)
Comment (romanetar) [reply src/actions/ranking-actions.js:104]: done

File: src/actions/selection-plan-actions.js [hunk 14]
```diff
@@ -516,4 +516,56 @@ export const deleteEventTypeSelectionPlan = (selectionPlanId, eventTypeId) => (d
             dispatch(stopLoading());
         }
     );
-}
\ No newline at end of file
+}
+
+/***********************  RATING TYPES  *******************************************/
+
+export const SELECTION_PLAN_RATING_TYPE_ADDED = 'SELECTION_PLAN_RATING_TYPE_ADDED';
```
[the remark this answers is not in the document]
Comment (romanetar) [reply src/actions/selection-plan-actions.js:523]: fixed

**Comment by romanetar** [comment]
@smarcet everything fixed as discussed by slack

**Pull request merged**